amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala

```diff
@@ -0,0 +1,48 @@
+package amf.aml.internal.parse.dialects.nodemapping.like
+
+import amf.aml.client.scala.model.domain.AnyMapping
+import amf.aml.internal.metamodel.domain.AnyMappingModel
+import amf.aml.internal.parse.dialects.DialectContext
+import amf.aml.internal.validate.DialectValidations.DialectError
+import amf.core.client.scala.model.domain.{AmfArray, AmfScalar}
+import amf.core.internal.metamodel.Field
+import amf.core.internal.parser.YMapOps
+import amf.core.internal.parser.domain.Annotations
+import org.yaml.model._
+
+trait AnyMappingParser {
+
+  def parse(map: YMap, mapping: AnyMapping)(implicit ctx: DialectContext): AnyMapping = {
+
+    processAnyMappingField(map, mapping, "allOf", AnyMappingModel.And)
+    processAnyMappingField(map, mapping, "oneOf", AnyMappingModel.Or)
+    processAnyMappingField(map, mapping, "components", AnyMappingModel.Components)
+    mapping
+
+  }
+
+  private def processAnyMappingField(map: YMap, mapping: AnyMapping, key: String, field: Field)(
+      implicit ctx: DialectContext): Unit = map.key(
+      key,
+      entry => {
+        entry.value.tagType match {
+          case YType.Seq =>
+            try {
+              val nodes = entry.value.as[YSequence].nodes.map(n => AmfScalar(n.as[String], Annotations(n)))
+              mapping.set(field, AmfArray(nodes, Annotations(entry.value)), Annotations(entry))
+            } catch {
+              case _: Exception =>
```

Comment:
Why are we managing exceptions here ? We don't manage exceptions in the rest of parsing code

Reply:
Deleted, the "as" error should be caught by the context